src/components/modals/generation/RankTwo.vue

```diff
@@ -0,0 +1,419 @@
+<script setup lang="ts">
+import { DEFAULT_NUMBER_OF_COMPONENTS } from "@/constants";
+import ExtendedMonzo from "@/monzo";
+import { parseLine } from "@/parser";
+import Scale from "@/scale";
+import {
+  makeRank2FromVals,
+  makeRank2FromCommas,
+  mosPatternsRank2FromVals,
+  mosPatternsRank2FromCommas,
+  type Rank2Params,
+} from "@/tempering";
+import { mosPatterns as getMosPatterns } from "moment-of-symmetry";
+import { computed, ref, watch } from "vue";
+import Modal from "@/components/ModalDialog.vue";
+import { makeState } from "@/components/modals/tempering-state";
+import { computedAndError } from "@/utils";
+import ScaleLineInput from "@/components/ScaleLineInput.vue";
+import { Interval } from "@/interval";
+
+const props = defineProps<{
+  centsFractionDigits: number;
+}>();
+
+const MAX_SIZE = 128;
```

Comment:
The max size is not respected when the number of periods is more than one. Need to fix in `moment-of-symmetry`.